libs/auth-api-lib/src/lib/services/delegations.service.ts

```diff
@@ -137,35 +147,38 @@ export class DelegationsService {
     return this.findOne(nationalId, id)
   }
 
-  async findOne(nationalId: string, id: string): Promise<Delegation | null> {
+  async findOne(nationalId: string, id: string): Promise<DelegationDTO | null> {
     this.logger.debug(`Finding a delegation with id ${id}`)
-    return await this.delegationModel.findOne({
+    const delegation = await this.delegationModel.findOne({
       where: {
         id: id,
         fromNationalId: nationalId,
       },
       include: [DelegationScope],
     })
+    return delegation ? delegation.toDTO() : null
   }
 
-  async findAllCustomTo(nationalId: string): Promise<Delegation[] | null> {
+  async findAllCustomTo(nationalId: string): Promise<DelegationDTO[] | null> {
     this.logger.debug(`Finding a delegation for nationalId ${nationalId}`)
-    return await this.delegationModel.findAll({
+    const delegations = await await this.delegationModel.findAll({
       where: {
         toNationalId: nationalId,
       },
       include: [DelegationScope],
     })
+    return delegations.map((delegation) => delegation.toDTO())
   }
 
-  async findAllCustomFrom(nationalId: string): Promise<Delegation[] | null> {
+  async findAllCustomFrom(nationalId: string): Promise<DelegationDTO[] | null> {
     this.logger.debug(`Finding a delegation for nationalId ${nationalId}`)
-    return await this.delegationModel.findAll({
+    const delegations = await this.delegationModel.findAll({
       where: {
         fromNationalId: nationalId,
       },
       include: [DelegationScope],
     })
+    return delegations.map((delegation) => delegation.toDTO())
```

Comment:
Shouldn't the controller be the one that maps the entity (and strip unwanted data) to DTO, since internally (i.e. between controller to service) you can just work with the original object?

Reply:
Good question. Opinions differ on how far up DTOs should go and how far down connected ORM objects should go. E.g. we commonly pass input DTO's to the service.

In this case, we're sharing this service between two different APIs (and corresponding controllers) and want to reuse as much implementation as possible between them. The DTO object provides a fine abstraction for the three different delegation sources returned by this service.